src/validators/resolve/validateExtensions.js

```diff
@@ -0,0 +1,25 @@
+import _ from 'lodash'
+import {friendlyTypeOf} from '../../utils'
+
+export default function validateExtensions(extensions) {
+  const type = friendlyTypeOf(extensions)
+
+  if (type !== 'array') {
+    return `Must be an array`
+  }
+
+  if (!extensions.includes('')) {
+    return `The extensions array should contain "", ` +
+       `so that modules that were required with their extension are properly resolved`
+  }
+
+  if (_.some(extensions, extension => friendlyTypeOf(extension) !== 'string')) {
```

Comment:
Could we assign the result of this expression to a variable? I feel like giving it a name helps keep the code self-documenting